test/OrchardCore.Tests/Scripting/ScriptFunctionsTest.cs

```diff
@@ -0,0 +1,112 @@
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using OrchardCore.Scripting;
+using OrchardCore.Tests.Apis.Context;
+using OrchardCore.Users;
+using OrchardCore.Users.Models;
+
+namespace OrchardCore.Tests.Scripting;
+public class ScriptFunctionsTest
+{
+    [Fact]
+    public async Task ProcessUserInfoTest()
+    {
+        using var context = new SiteContext();
+        await context.InitializeAsync();
+        await context.UsingTenantScopeAsync(async scope =>
+        {
+            var userProperties = @"{
+            ""UserProfile"": {
+                ""ContentItemId"": ""4fkpnj0fnawmzy3zdc5kx5pnn1"",
+                ""ContentItemVersionId"": null,
+                ""ContentType"": ""UserProfile"",
+                ""DisplayText"": ""admin admin - (admin)"",
+                ""Latest"": true,
+                ""Published"": true,
+                ""ModifiedUtc"": null,
+                ""PublishedUtc"": null,
+                ""CreatedUtc"": null,
+                ""Owner"": ""4fkpnj0fnawmzy3zdc5kx5pnn1"",
+                ""Author"": ""admin"",
+                ""UserProfile"": {
+                    ""UserName"": {
+                        ""Text"": ""admin""
+                    },
+                    ""Email"": {
+                        ""Text"": ""admin@admin.com""
+                    },
+                    ""OwnerUser"": {
+                        ""UserIds"": [
+                            ""4fkpnj0fnawmzy3zdc5kx5pnn1""
+                        ],
+                        ""UserNames"": []
+                    },
+                    ""DisplayName"": {
+                        ""Text"": ""admin admin""
+                    },
+                    ""Department"": {
+                        ""ContentItemIds"": []
+                    },
+                    ""Manager"": {
+                        ""UserIds"": [],
+                        ""UserNames"": []
+                    },
+                    ""FirstName"": {
+                        ""Text"": null
+                    },
+                    ""LastName"": {
+                        ""Text"": null
+                    },
+                    ""Country"": {
+                        ""Text"": ""ID""
+                    },
+                    ""Countries"": {
+                        ""Values"": []
+                    }
+                },
+                ""TitlePart"": {
+                    ""Title"": ""admin - (admin)""
+                },
+                ""DIndexPart"": {}
+            },
+            ""UserNotificationPreferencesPart"": {
+                ""Methods"": [
+                    ""Email""
+                ],
+                ""Optout"": []
+            },
+            ""UserTimeZone"": {
+                ""TimeZoneId"": null
+            }
+        }";
+
+            var userManager = scope.ServiceProvider.GetRequiredService<UserManager<IUser>>();
```

Comment:
Why do you need to fetch a user and then remove properties? During arrange, manually create a user with some key values to test instead.